hoplite-core/src/test/kotlin/com/sksamuel/hoplite/SystemPropertiesPropertySourceTest.kt

```diff
@@ -0,0 +1,59 @@
+package com.sksamuel.hoplite
+
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.maps.shouldContainExactly
+import io.kotest.matchers.shouldBe
+import java.util.Properties
+class SystemPropertiesPropertySourceTest : FunSpec() {
+  init {
+    test("reads config from string") {
+      data class TestConfig(val a: String, val b: Int)
+
+      val config = ConfigLoader {
+        withDefaultSources(false)
+        addPropertySource(
+          SystemPropertiesPropertySource(
+            systemPropertiesMap = {
+              mapOf("config.override.a" to "Overridden by system prop")
+            },
+          )
+        )
+        addPropertySource(
+          PropertySource.string(
+            """
+          a = A value
+          b = 42
+          """.trimIndent(), "props"
+          )
+        )
+      }.loadConfigOrThrow<TestConfig>()
+
+      config shouldBe TestConfig("Overridden by system prop", 42)
+    }
+
+    test("reads from real system props") {
+      data class TestConfig(val a: String, val b: Int)
+
+      try {
+        // this is an error prone test because we have to remember to clear these properties after the test
+        System.setProperty("config.override.a", "Overridden by system prop")
+        System.setProperty("config.override.b", "42")
+
+        // use default property sources to load system properties
+        ConfigLoader().loadConfigOrThrow<TestConfig>() shouldBe TestConfig("Overridden by system prop", 42)
+      } finally {
+        System.clearProperty("config.override.a")
+        System.clearProperty("config.override.b")
+      }
```

Comment:
This is the type of test that trying to avoid in general, but in this case, this shows that the default config still picks up the "real" system properties.

Reply:
Yep good idea.